cmd/frontend/registry/api/featured_extensions.go

```diff
@@ -0,0 +1,49 @@
+package api
+
+import (
+	"context"
+	"sync"
+
+	"github.com/sourcegraph/sourcegraph/cmd/frontend/graphqlbackend"
+	"github.com/sourcegraph/sourcegraph/internal/database/dbutil"
+)
+
+type featuredExtensionsResolver struct {
+	// cache results because they are used by multiple fields
+	once sync.Once
+
+	featuredExtensions []graphqlbackend.RegistryExtension
+	err                error
+	db                 dbutil.DB
+}
+
+func (r *extensionRegistryResolver) FeaturedExtensions(ctx context.Context) (graphqlbackend.FeaturedExtensions, error) {
+	return &featuredExtensionsResolver{db: r.db}, nil
+}
+
+func (r *featuredExtensionsResolver) compute(ctx context.Context) ([]graphqlbackend.RegistryExtension, error) {
+	r.once.Do(func() {
+		xs, err := GetFeaturedExtensions(ctx, r.db)
+		if err != nil {
+			r.err = err
```

Comment:
```suggestion
			r.err = err
			return
```

This is not required in this case, but good practice. Otherwise, this function can be simplified as:

```go
r.featuredExtensions, r.err = GetFeaturedExtensions(ctx, r.db)
```